### PR TITLE
Revise asin op

### PR DESCRIPTION
**Pull request by bszmelcz**
Add type_prop test for asin operator

ticket: 37420

### CODE REVIEW

File: ngraph/test/type_prop/asin.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+{
+    const PartialShape param_shape{Dimension::dynamic(), 10};
+    const auto param = make_shared<op::Parameter>(element::f32, param_shape);
+    const auto op = make_shared<op::Asin>(param);
+    ASSERT_TRUE(op->get_output_partial_shape(0).same_scheme(PartialShape{Dimension(), 10}));
+}
+
+TEST(type_prop, asin_dynamic_rank_input_3D)
+{
+    const PartialShape param_shape{2, Dimension::dynamic(), Dimension::dynamic()};
```
Comment (ilyachur) [inline ngraph/test/type_prop/asin.cpp:49]: Can we add tests with dynamic rank?

Comment (bszmelcz) [reply ngraph/test/type_prop/asin.cpp:49]: sure, will address that today

**Review by pelszkow — APPROVED** [review]

**Comment by ilyachur** [comment]
@bszmelcz Should we close this PR? I guess PR https://github.com/openvinotoolkit/openvino/pull/3815 does the same thing.

**Comment by bszmelcz** [comment]
> @bszmelcz Should we close this PR? I guess PR #3815 does the same thing.

yes, after adding the new class for unary operators I thought that new PR will be cleaner instead of making a whole change here, moreover this one is pretty old.

**Pull request closed without merging**